extensions/wikia/Email/Controller/FounderController.class.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+namespace Email\Controller;
+
+use Email\Check;
+use Email\EmailController;
+use Email\Fatal;
+
+abstract class FounderController extends EmailController {
+
+	/** @var \Title */
+	protected $pageTitle;
+
+	protected $previousRevId;
+	protected $currentRevId;
+
+	public function initEmail() {
+		// This title is for the article being commented upon
+		$titleText = $this->request->getVal( 'pageTitle' );
+		$titleNamespace = $this->request->getVal( 'pageNs' );
+
+		$this->pageTitle = \Title::newFromText( $titleText, $titleNamespace );
+
+		$this->previousRevId = $this->request->getVal( 'previousRevId' );
+		$this->currentRevId = $this->request->getVal( 'currentRevId' );
+
+		$this->assertValidParams();
+	}
+
+	/**
+	 * Validate the params passed in by the client
+	 */
+	private function assertValidParams() {
+		$this->assertValidTitle();
+		$this->assertValidRevisionIds();
+	}
+
+	/**
+	 * @throws \Email\Check
+	 */
+	private function assertValidTitle() {
+		if ( !$this->pageTitle instanceof \Title ) {
+			throw new Check( "Invalid value passed for title" );
+		}
+
+		if ( !$this->pageTitle->exists() ) {
+			throw new Check( "Title doesn't exist." );
+		}
+	}
+
+	/**
+	 * @throws \Email\Check
+	 */
+	private function assertValidRevisionIds() {
+		if ( empty( $this->previousRevId ) ) {
+			throw new Check( "Invalid value passed for previousRevId" );
+		}
+
+		if ( empty( $this->currentRevId ) ) {
+			throw new Check( "Invalid value passed for currentRevId" );
+		}
+	}
+
+	/**
+	 * @template avatarLayout
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'editorProfilePage' => $this->getCurrentProfilePage(),
+			'editorUserName' => $this->getCurrentUserName(),
+			'editorAvatarURL' => $this->getCurrentAvatarURL(),
+			'summary' => $this->getSummary(),
+			'buttonText' => $this->getChangesLabel(),
+			'buttonLink' => $this->getChangesLink(),
+			'contentFooterMessages' => [
+				$this->getFooterEncouragement(),
+				$this->getFooterArticleLink(),
+				$this->getFooterAllChangesLink(),
+			],
+			'details' => $this->getDetails(),
+			'hasContentFooterMessages' => true
```

Comment:
Yea, I spent some time on another email ticket trying to pull out this `hasContentFooterMessages` variable, but there's no way in mustache to use the length of an array has a boolean. Or at least when you do, it won't loop over the array.

Damian is doing something similar to what you're suggesting @mklucsarits [here](https://github.com/Wikia/app/blob/dev/extensions/wikia/Email/Controller/WatchedPageController.class.php#L92). I think the difference is that the check is done in the base class, and then some of the sublcasses will have contentFooterMessages and some won't. In this case where we always have contentFooterMessages I think just using `true` makes sense.